torch/fx/experimental/Partitioner.py

```diff
@@ -0,0 +1,151 @@
+from torch.fx import Graph, GraphModule
+from torch.fx import Node
+from typing import Dict, List, Tuple
+"""a graph module may not fit into one backend.
+Partitioner class helps cut one graph into subgraphs (partitions),
+so that each partition could fit into different backends
+"""
+class Partition:
+    def __init__(self, partition_id: int) -> None:
+        self.parents: List[Partition] = []
+        self.children: List[Parttion] = []
+        """
+           One partion only has to fit into one device
+           'backend type' is the type of that device
+        """
+        self.partition_id = partition_id
+        """
+           self.node_list contains graph nodes that will be deployed in this partition
+           List[[node_index, node_name, node_op]]
+           with node_ndex, it is easy to track if any node is missing from all partitions
+           node_name is to show what this node does
+           node_op is used to find placeholders for input node of the partition
+        """
+        self.node_list: List[[Tuple[int, str, str]]] = []
+
+    def add_node(self, node_index: int, node_name: str, node_op: str) -> None:
+        """append a new node into the partition"""
```

Comment:
general nit: Try to begin your comments with a capital letter and end with a period.
-- "Append a new node into the partition."